### PR TITLE
[datavalues] add unit test for scatter 

### PR DESCRIPTION
**Pull request by jyizheng**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Add unit tests for   common/datavalues/src/arrays/ops/scatter.rs

## Changelog

- Build/Testing/CI

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
It would better to add some docs for the `scatter_unchecked` trait:
https://github.com/datafuselabs/datafuse/bl/668e1854a3b1e62d5dd3b23cb2539ad55a40f504/common/datavalues/src/arrays/ops/scatter.rs#L23-L27

**Comment by jyizheng** [comment]
> It would better to add some docs for the `scatter_unchecked` trait:
> https://github.com/datafuselabs/datafuse/bl/668e1854a3b1e62d5dd3b23cb2539ad55a40f504/common/datavalues/src/arrays/ops/scatter.rs#L23-L27

Done!

**Review by sundy-li — APPROVED** [review]

File: common/datavalues/src/arrays/ops/scatter.rs [hunk 1]
```diff
@@ -20,6 +20,16 @@ use crate::*;
...
+
+    /// scatter_unchecked() partitions the input array into multiple arrays.
+    /// _indices: an iterateor of vector whose length is the same as the array.
+    /// The element of _indices indicates which group the corresponding row
+    /// in the input array belongs to.
+    /// _scattered_size: the number of partitions
+    ///
+    /// Example: if the input array has five rows [1, 2, 3, 4] and
+    /// _indices = [0, 1, 0, 1] and _scatter_size = 2,
+    /// then the output would be a vector of two arrays: [1, 4] and [2, 3].
```
Comment (zhang2014) [inline common/datavalues/src/arrays/ops/scatter.rs:32]: [1, 3] and [2, 4]

Comment (BohuTANG) [reply common/datavalues/src/arrays/ops/scatter.rs:32]: https://github.com/datafuselabs/datafuse/issues/1195
Expect take by the community.

**Pull request merged**